content/en/docs/faq/installation/uninstall-pluggable-components.md

```diff
@@ -0,0 +1,215 @@
+   do
+     kubectl patch devopsprojects $devopsproject -p '{"metadata":{"finalizers":null}}' --type=merge
+   done
+   
+   for pip in `kubectl get pipeline -A -o jsonpath="{.items[*].metadata.name}"`
+   do
+     kubectl patch pipeline $pip -n `kubectl get pipeline -A | grep $pip | awk '{print $1}'` -p '{"metadata":{"finalizers":null}}' --type=merge
+   done
+   
+   for s2ibinaries in `kubectl get s2ibinaries -A -o jsonpath="{.items[*].metadata.name}"`
+   do
+     kubectl patch s2ibinaries $s2ibinaries -n `kubectl get s2ibinaries -A | grep $s2ibinaries | awk '{print $1}'` -p '{"metadata":{"finalizers":null}}' --type=merge
+   done
+   
+   for s2ibuilders in `kubectl get s2ibuilders -A -o jsonpath="{.items[*].metadata.name}"`
+   do
+     kubectl patch s2ibuilders $s2ibuilders -n `kubectl get s2ibuilders -A | grep $s2ibuilders | awk '{print $1}'` -p '{"metadata":{"finalizers":null}}' --type=merge
+   done
+   
+   for s2ibuildertemplates in `kubectl get s2ibuildertemplates -A -o jsonpath="{.items[*].metadata.name}"`
+   do
+     kubectl patch s2ibuildertemplates $s2ibuildertemplates -n `kubectl get s2ibuildertemplates -A | grep $s2ibuildertemplates | awk '{print $1}'` -p '{"metadata":{"finalizers":null}}' --type=merge
+   done
+   
+   for s2iruns in `kubectl get s2iruns -A -o jsonpath="{.items[*].metadata.name}"`
+   do
+     kubectl patch s2iruns $s2iruns -n `kubectl get s2iruns -A | grep $s2iruns | awk '{print $1}'` -p '{"metadata":{"finalizers":null}}' --type=merge
+   done
+   
+   kubectl delete devopsprojects --all 2>/dev/null
+   ```
+
+   ```bash
+   kubectl delete ns kubesphere-devops-system
+   ```
+
+## Uninstall KubeSphere Logging
+
+1. Change the value of `logging.enabled` from `true` to `false` in `ks-installer`.
+
+2. To disable only log collection:
+
+   ```bash
+   delete inputs.logging.kubesphere.io -n kubesphere-logging-system tail
+   ```
+
+   {{< notice note >}}
+
+   After running this command, KubeSphere stops collecting container logs.
+
+   {{</ notice >}}
+
+3. To uninstall Logging system including Elasticsearch:
+
+   ```bash
+   kubectl delete crd fluentbitconfigs.logging.kubesphere.io
+   kubectl delete crd fluentbits.logging.kubesphere.io
+   kubectl delete crd inputs.logging.kubesphere.io
+   kubectl delete crd outputs.logging.kubesphere.io
+   kubectl delete crd parsers.logging.kubesphere.io
+   kubectl delete deployments.apps -n kubesphere-logging-system fluentbit-operator
+   helm uninstall elasticsearch-logging --namespace kubesphere-logging-system
+   ```
+
+   {{< notice note >}}
+
+   This operation may cause anomalies in Auditing, Events, and Service Mesh.
+
+   {{</ notice >}}
+
+## Uninstall KubeSphere Events
+
+1. Change the value of `events.enabled` from `true` to `false` in `ks-installer`.
+
+2. Run the following command:
+
+   ```bash
+   helm delete ks-events -n kubesphere-logging-system
+   ```
+
+## Uninstall KubeSphere Alerting and Notification
```

Comment:
Perfom the following steps to uninstall Alerting for KubeSphere v3.1.x: 
1. Change the value of `alerting.enabled` from `true` to `false` in `ks-installer`.
2. Run the following command:
    ```bash
    kubectl -n kubesphere-monitoring-system delete thanosruler kubesphere
    ```
In addition, Notification is installed by default and may no longer be enabled or disabled

Reply:
Thanks for your comment. Updates made in [this commit](https://github.com/kubesphere/website/pull/1863/commits/d48ea46b0fb97537a354d11ef92671abb5f35f5f).